File: app/src/main/java/com/example/shenjack/zhihudailyreader/stoylist/StoryListContract.java
package com.example.shenjack.zhihudailyreader.stoylist;

import com.example.shenjack.zhihudailyreader.BaseView;
import com.example.shenjack.zhihudailyreader.data.StoriesBean;

import java.util.List;

/**
 * Created by ShenJack on 2017/5/29.
 */

public interface StoryListContract {

    interface View extends BaseView<Presenter>{
        void showLoadingIndicator(boolean visible);

        void showStories(List<StoriesBean> stories);

        void showStoryDetail(StoriesBean story);

        void showLoadingError();

        void showDateSelector();


        void setStories(List<StoriesBean> mStories);

        String getDate();

        void notifyLoadingComplete();

        void notifyLoadingError();


    }

    interface Presenter {
        void getTodayStories(boolean forceUpdate);

        void fetchBeforeStories(String date);

        void loadStories(String data);

        void openStoryDetail(int storyId);

        void refreshStoryList();

    }
}

File: app/src/main/java/com/example/shenjack/zhihudailyreader/stoylist/StoryListContract.kt
package com.example.shenjack.zhihudailyreader.stoylist

import com.example.shenjack.zhihudailyreader.BaseView
import com.example.shenjack.zhihudailyreader.data.StoriesBean

/**
 * Created by ShenJack on 2017/5/29.
 */

interface StoryListContract {

    interface View : BaseView<Presenter> {
        fun showLoadingIndicator(visible: Boolean)

        fun showStories(stories: List<StoriesBean>)

        fun showStoryDetail(story: StoriesBean)

        fun showLoadingError()

        fun showDateSelector()

        var date:String

        fun setStories(stories: List<StoriesBean>)


        fun notifyLoadingComplete()

        fun notifyLoadingError()

    }

    interface Presenter {
        fun getTodayStories(forceUpdate: Boolean)

        fun fetchBeforeStories(date: String)

        fun loadStories(data: String)

        fun openStoryDetail(storyId: Int, title: String)

        fun refreshStoryList()

    }
}
